File: go-service/models/delivery.go
package models

import "time"

// Delivery represents a delivery order
type Delivery struct {
	ID              int        `db:"Id" json:"id"`
	DeliveryNumber  string     `db:"DeliveryNumber" json:"deliveryNumber"`
	SaleID          *int       `db:"SaleId" json:"saleId,omitempty"`
	CustomerID      int        `db:"CustomerId" json:"customerId"`
	DriverID        int        `db:"DriverId" json:"driverId"`
	DeliveryAddress string     `db:"DeliveryAddress" json:"deliveryAddress"`
	ContactPhone    *string    `db:"ContactPhone" json:"contactPhone,omitempty"`
	Status          string     `db:"Status" json:"status"`
	Priority        string     `db:"Priority" json:"priority"`
	ScheduledDate   time.Time  `db:"ScheduledDate" json:"scheduledDate"`
	DeliveredDate   *time.Time `db:"DeliveredDate" json:"deliveredDate,omitempty"`
	Notes           *string    `db:"Notes" json:"notes,omitempty"`
	CreatedAt       time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt       *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

// DeliveryItem represents an item in a delivery
type DeliveryItem struct {
	ID         int       `db:"Id" json:"id"`
	DeliveryID int       `db:"DeliveryId" json:"deliveryId"`
	ProductID  int       `db:"ProductId" json:"productId"`
	Quantity   int       `db:"Quantity" json:"quantity"`
	CreatedAt  time.Time `db:"CreatedAt" json:"createdAt"`
}

File: frontend/src/app/core/domain/delivery/delivery.ts
import { DeliveryStatus, Priority } from '../../enums/enums.barrel';
import { Sale } from '../sale/sale';
import { Customer } from '../customer/customer';
import { Driver } from '../driver/driver';

/**
 * Represents a Delivery Item
 */
export interface DeliveryItem {
  /** ID */
  id: number;

  /** Delivery ID */
  deliveryId: number;

  /** Product ID */
  productId: number;

  /** Quantity */
  quantity: number;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}

/**
 * Represents a Delivery
 */
export interface Delivery {
  /** ID */
  id: number;

  /** Delivery Number - unique identifier */
  deliveryNumber: string;

  /** Sale ID - optional FK to Sale */
  saleId?: number | null;

  /** Customer ID - FK to Customer */
  customerId: number;

  /** Driver ID - FK to Driver */
  driverId: number;

  /** Delivery Address */
  deliveryAddress: string;

  /** Contact Phone */
  contactPhone?: string | null;

  /** Status */
  status: DeliveryStatus;

  /** Priority */
  priority: Priority;

  /** Scheduled Date */
  scheduledDate: Date;

  /** Delivered Date */
  deliveredDate?: Date | null;

  /** Notes */
  notes?: string | null;

  /** Customer - associated customer */
  customer?: Customer;

  /** Driver - assigned driver */
  driver?: Driver;

  /** Sale - associated sale (if any) */
  sale?: Sale; // Sale type

  /** Items - delivery items */
  items?: DeliveryItem[];

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
